src/utilities/middleware-utils.ts

```
@@ -0,0 +1,278 @@
+): Promise<{ result: T; duration: number }> {
+  return new Promise(async (resolve) => {
+    const start = performance.now();
+    const result = await operation();
+    const duration = performance.now() - start;
+
+    if (process.env.NODE_ENV === 'development') {
+      console.log(`${label}: ${duration.toFixed(2)}ms`);
+    }
+
+    resolve({ result, duration });
+  });
```

Biome (2.1.2)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)